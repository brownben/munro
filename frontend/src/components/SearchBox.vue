<template>
  <form
    class="relative w-full sm:w-1/2 lg:w-1/3"
    @submit.prevent="go(`/search/${query}`)"
  >
    <label for="searchBox" class="sr-only">Search Query:</label>
    <input
      id="searchBox"
      v-model="query"
      type="text"
      spellcheck="false"
      class="
        w-full
        py-2
        pl-10
        pr-24
        font-sans
        leading-snug
        text-gray-500
        transition
        duration-300
        ease-in-out
        border border-gray-100
        outline-none
        appearance-none
        bg-gray-50
        rounded-shape
        focus:border-main-300
        focus:text-gray-900
        focus:bg-white
        focus:shadow-outline
      "
      @focus="searchFocused = true"
      @blur="searchFocused = false"
    />
    <button
      class="
        absolute
        top-0
        right-0
        px-4
        py-2
        transition
        duration-300
        outline-none
        font-heading
        rounded-shape
      "
      :class="
        searchFocused
          ? 'bg-main-400 text-white hover:bg-main-500 focus:bg-main-500 hover:text-white focus:text-white'
          : 'text-gray-600 bg-gray-100 hover:bg-main-200 focus:bg-main-200 hover:text-main-700 focus:text-main-700'
      "
      @click="go(`/search/${query}`)"
    >
      Search
    </button>
    <svg
      viewBox="0 0 20 20"
      fill="currentColor"
      class="absolute top-0 left-0 w-6 h-6 py-px my-2 ml-2"
      :class="searchFocused ? 'text-main-500' : 'text-gray-400'"
      aria-hidden="true"
    >
      <path
        fill-rule="evenodd"
        d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
        clip-rule="evenodd"
      />
    </svg>
  </form>
</template>

<script setup lang="ts">
import { ref } from 'vue'
import { useRouter, useRoute } from 'vue-router'

const router = useRouter()
const route = useRoute()

const searchFocused = ref(false)
const query = ref(route.params.query)

const go = (location: string) => {
  if (location !== route.path) router.push(location)
}
</script>
