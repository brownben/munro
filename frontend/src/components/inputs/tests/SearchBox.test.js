import { mount } from '@vue/test-utils'
import SearchBox from '@/components/inputs/SearchBox.vue'

test('Is a Vue Instance', () => {
  const wrapper = mount(SearchBox, {
    stubs: ['router-link'],
    mocks: {
      $router: { push: jest.fn() },
      $route: { params: { query: '' } },
    },
  })
  expect(wrapper.isVueInstance()).toBeTruthy()
})

test('Renders Correctly', () => {
  const wrapper = mount(SearchBox, {
    stubs: ['router-link'],
    mocks: {
      $router: { push: jest.fn() },
      $route: { params: { query: '' } },
    },
  })
  expect(wrapper.element).toMatchSnapshot()
})

test('Go', () => {
  const wrapper = mount(SearchBox, {
    stubs: ['router-link'],
    mocks: {
      $router: { push: jest.fn() },
      $route: { path: 'NOT', params: { query: '' } },
    },
  })
  wrapper.vm.go('NOT')
  expect(wrapper.vm.$router.push).not.toHaveBeenCalledWith('NOT')
  wrapper.vm.go('a')
  expect(wrapper.vm.$router.push).toHaveBeenCalledWith('a')
})
