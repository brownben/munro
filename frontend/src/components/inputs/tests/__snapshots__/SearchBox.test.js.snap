// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<form
  class="relative w-full sm:w-1/2 lg:w-1/3"
>
  <label
    class="sr-only"
    for="searchBox"
  >
    Search Query:
  </label>
   
  <input
    class="w-full py-2 pl-10 pr-24 font-sans leading-snug text-gray-500 transition duration-300 ease-in-out border border-gray-100 outline-none appearance-none bg-gray-50 rounded-shape focus:border-main-300 focus:text-gray-900 focus:bg-white"
    id="searchBox"
    spellcheck="false"
    type="text"
  />
   
  <button
    class="absolute top-0 right-0 px-4 py-2 transition duration-300 outline-none font-heading rounded-shape text-gray-600 bg-gray-100 hover:bg-main-200 focus:bg-main-200 hover:text-main-700 focus:text-main-700"
  >
    
    Search
  
  </button>
   
  <svg
    class="absolute top-0 left-0 w-6 h-6 py-px my-2 ml-2 text-gray-400"
    fill="currentColor"
    viewBox="0 0 20 20"
  >
    <path
      clip-rule="evenodd"
      d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
      fill-rule="evenodd"
    />
  </svg>
</form>
`;
